DiscordChatExporter.Gui/ViewModels/RootViewModel.cs

```diff
@@ -236,6 +237,25 @@ public async void ExportChannels()
             // Notify of overall completion
             if (successfulExportCount > 0)
                 Notifications.Enqueue($"Successfully exported {successfulExportCount} channel(s)");
+
+            IPartitioner CreatePartitioner()
```

Comment:
Maybe `PartitionFormat` should implement an enum class pattern and have a `CreatePartitioner` method to avoid this if-else chain. However, if we don't plan on adding any more partition formats, this seems like extra work for nothing.

Also, why do I need these `(int)` casts near the end? They are guarded by a previous if-return that checks if `partitionLimit` is null. I thought the C# compiler could recognize this?

Reply:
Sorry, I thought I selected every line of `CreatePartitoner` to mention in this comment, but I guess I don't know how to do that XD

Reply:
> Also, why do I need these (int) casts near the end? They are guarded by a previous if-return that checks if partitionLimit is null. I thought the C# compiler could recognize this?

C# compiler is generally pretty dumb. It doesn't have the level of flow analysis that TS has, for example.

> Sorry, I thought I selected every line of CreatePartitoner to mention in this comment, but I guess I don't know how to do that XD

NP. To do it, you need to click and hold.

Reply:
You can, however, streamline it like this:

```csharp
return (partitionFormat, partitionLimit) switch
{
   (PartitionFormat.MessageCount, int messageLimit) => new MessageCountPartitioner(messageLimit),
   (PartitionFormat.FileSize, int fileSizeLimit) => new FileSizePartitioner(fileSizeLimit),
   _ => new NullPartitioner()
}
```

(my syntax may be wrong, don't exactly remember how to write this from memory)

Reply:
Very cool